### PR TITLE
MLIBZ-3066: Kotlin Network

### PR DESCRIPTION
**Pull request by heyzooi**
#### Description

Migration to Kotlin for network classes

#### Changes

- Network related classes migrated to Kotlin

#### Tests

- Same tests

### CODE REVIEW

**Comment by b-stolyarov** [comment]
Should we merge this PR after merging `kotlin-migration` to master?

**Comment by vinaygahlawat** [comment]
@b-stolyarov Let's wait to resolve the issue with running unit tests before merging.

File: java-api-core/java-api-core.iml [hunk 1]
```diff
@@ -12,48 +12,98 @@
...
...24ba069a09e964ad2640c36f140f2c758/kotlin-stdlib-jdk7-1.3.41.jar:/Users/victor/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.3.41/e24bd38de28a326cce8b1f0d61e809e9a92dad6a/kotlin-stdlib-1.3.41.jar:/Users/victor/.gradle/caches/modules-2/files-2.1/org.apache.httpcomponents/httpcore/4.0.1/e813b8722c387b22e1adccf7914729db09bcb4a9/httpcore-4.0.1.jar:/Users/victor/.gradle/caches/modules-2/files-2.1/commons-logging/commons-logging/1.1.1/5043bfebc3db072ed80fbd362e7caf00e885d8ae/commons-logging-1.1.1.jar:/Users/victor/.gradle/caches/modules-2/files-2.1/commons-codec/commons-codec/1.3/fd32786786e2adb664d5ecc965da47629dca14ba/commons-codec-1.3.jar:/Users/victor/.gradle/caches/modules-2/files-2.1/com.squareup.okio/okio/1.13.0/a9283170b7305c8d92d25aff02a6ab7e45d06cbe/okio-1.13.0.jar:/Users/victor/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.3.41/2ecf4aa059427d7186312fd1736afedf7972e7f7/kotlin-stdlib-common-1.3.41.jar:/Users/victor/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar:/Users/victor/dev/android-sdk/java-api-core/build/gen/buildconfig/classes/main" />
```
Comment (vinaygahlawat) [inline java-api-core/java-api-core.iml]: A lot of these file paths seem to be local (e.g. _**/Users/victor/...**_), why is that?

Comment (heyzooi) [reply java-api-core/java-api-core.iml]: not sure, the sync button from Android Studio should not add non-relative paths to this file. I will take a look on those paths

Comment (b-stolyarov) [reply java-api-core/java-api-core.iml]: It's generated by Android Studio files, I think they can be added to .gitignore files (java-api-core.iml, android-lib.iml, parent.iml). Usually, I just don't commit them.

Comment (heyzooi) [reply java-api-core/java-api-core.iml]: right, good call. I will remove them and add to the .gitignore file

**Pull request merged**